sdk/tests/conformance2/transform_feedback/too-small-buffers.html

```diff
@@ -0,0 +1,242 @@
+    ["out_value1", "out_value2"], gl.INTERLEAVED_ATTRIBS,
+    ["in_value1", "in_value2"]);
+const progSeparate = wtu.setupTransformFeedbackProgram(gl, ["vshader", "fshader"],
+    ["out_value1", "out_value2"], gl.SEPARATE_ATTRIBS,
+    ["in_value1", "in_value2"]);
+wtu.glErrorShouldBe(gl, gl.NO_ERROR, "program compilation");
+
+// Attrib 1 contains 4 vertices. Attrib 2 contains 4 instance indices.
+const vertexBuffer0 = gl.createBuffer();
+gl.bindBuffer(gl.ARRAY_BUFFER, vertexBuffer0);
+gl.bufferData(gl.ARRAY_BUFFER, new Float32Array([1, 2, 3, 4]), gl.STATIC_DRAW);
+const vertexBuffer1 = gl.createBuffer();
+gl.bindBuffer(gl.ARRAY_BUFFER, vertexBuffer1);
+gl.bufferData(gl.ARRAY_BUFFER, new Float32Array([1, 2, 3, 4]), gl.STATIC_DRAW);
+gl.bindBuffer(gl.ARRAY_BUFFER, vertexBuffer0);
+gl.enableVertexAttribArray(0);
+gl.vertexAttribPointer(0, 1, gl.FLOAT, false, 0, 0);
+gl.bindBuffer(gl.ARRAY_BUFFER, vertexBuffer1);
+gl.enableVertexAttribArray(1);
+gl.vertexAttribPointer(1, 1, gl.FLOAT, false, 0, 0);
+gl.vertexAttribDivisor(1, 1);
+
+let tfBuffer0 = gl.createBuffer();
+let tfBuffer1 = gl.createBuffer();
+
+wtu.glErrorShouldBe(gl, gl.NO_ERROR, "setup");
+
+let cases = [
+    { name: "drawArrays",
+      drawFunction: ()=>gl.drawArrays(gl.POINTS, 0, 4),
+      result: [[2, 4, 6, 8], [2, 2, 2, 2]]},
+    { name: "drawArraysInstanced one instance",
+      drawFunction: ()=>gl.drawArraysInstanced(gl.POINTS, 0, 4, 1),
+      result: [[2, 4, 6, 8], [2, 2, 2, 2]]},
+    { name: "drawArraysInstanced four instances",
+      drawFunction: ()=>gl.drawArraysInstanced(gl.POINTS, 0, 1, 4),
+      result: [[2, 2, 2, 2], [2, 4, 6, 8]]},
+  ];
+
+for (let {name, drawFunction, result} of cases) {
+  debug("<h1>" + name + "</h1>")
+
+  let interleavedResult = [];
+  for (let i = 0; i < result[0].length; i++) {
+    interleavedResult.push(result[0][i], result[1][i]);
+  }
+
+  let doTransformFeedback = (drawFunction, error) => {
+    gl.beginTransformFeedback(gl.POINTS);
+    wtu.glErrorShouldBe(gl, gl.NO_ERROR, "before draw");
+    drawFunction();
+    wtu.glErrorShouldBe(gl, error, "draw");
+    gl.endTransformFeedback();
+  }
+
+  gl.useProgram(progInterleaved);
+
+  debug("<h3>interleaved - Baseline success case</h3>")
+  gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 0, tfBuffer0);
+  gl.bufferData(gl.TRANSFORM_FEEDBACK_BUFFER, 8*4, gl.STREAM_READ);
+  doTransformFeedback(drawFunction, gl.NO_ERROR);
+  wtu.checkFloatBuffer(gl, gl.TRANSFORM_FEEDBACK_BUFFER, interleavedResult);
+
+  debug("<h3>interleaved - Buffer too small</h3>")
+  gl.bufferData(gl.TRANSFORM_FEEDBACK_BUFFER, 8*4-1, gl.STREAM_READ);
+  doTransformFeedback(drawFunction, gl.INVALID_OPERATION);
+  wtu.checkFloatBuffer(gl, gl.TRANSFORM_FEEDBACK_BUFFER,
+      [0, 0, 0, 0, 0, 0, 0]);
+
+  debug("<h3>interleaved - Multiple draws success case</h3>")
+  gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 0, tfBuffer0);
+  gl.bufferData(gl.TRANSFORM_FEEDBACK_BUFFER, 8*4*2, gl.STREAM_READ);
```

Comment:
Maybe I misunderstood something, but why do we need twice the data store? Won't the second draw simply overwrite the results from the first draw?

Same question for next case also.

Reply:
Per offline discussion with @jdarpinian , ES spec 3.0.4, page 89,

"When transform feedback is paused, no vertices are recorded. When transform
feedback is resumed, subsequent vertices are appended to the bound buffer objects
immediately following the last vertex written before transform feedback was
paused."

So it seems between draw calls within a transformfeedback session (between Begin* and End* calls), buffer pointer does advance and carry over to the next draw call.